File: SpringTILCode/src/main/java/com/example/springtilcode/domain/post/controller/PostController.java
package com.example.springtilcode.domain.post.controller;

import com.example.springtilcode.domain.post.controller.dto.request.PostRequest;
import com.example.springtilcode.domain.post.controller.dto.response.PostResponse;
import com.example.springtilcode.domain.post.service.PostService;
import lombok.RequiredArgsConstructor;
import org.springframework.web.bind.annotation.*;

@RestController
@RequiredArgsConstructor
public class PostController {
    private final PostService postService;

    @PostMapping("/post")
    public void save(@RequestBody PostRequest request){
        postService.save(request.getTitle(), request.getContent());
    }

    @DeleteMapping("/post/{id}")
    public void delete(@PathVariable Integer id){
        postService.delete(id);
    }

    @GetMapping("/post/{id}")
    public PostResponse getPost(@PathVariable Integer id){
        return postService.getPost(id);
    }
}

File: kotlin/src/main/kotlin/com/example/demo/domain/feed/controller/PostController.kt
package com.example.demo.domain.feed.controller

import com.example.demo.domain.feed.controller.dto.request.PostRequest
import com.example.demo.domain.feed.controller.dto.response.PostResponse
import com.example.demo.domain.feed.service.PostService
import org.springframework.web.bind.annotation.DeleteMapping
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.PathVariable
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RestController

@RestController
class PostController (
    private val postService: PostService
){
    @PostMapping("/post")
    fun create(@RequestBody request: PostRequest): PostResponse{
        return postService.create(request)
    }

    @DeleteMapping("/post/{id}")
    fun delete(@PathVariable id: Long){
        postService.delete(id)
    }

    @GetMapping("/post/{id}")
    fun getPost(@PathVariable id: Long): PostResponse{
        return postService.getFeed(id)
    }

}